pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDaoImpl.java

```
@@ -515,6 +524,58 @@ private boolean booleanCheckFoundAsName(String kitLabel, String query) {
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_SELECT_KIT_REQUEST
+                    + KIT_BY_HRUID_AND_DDP_LABEL.replace("#", hruid))) {
+                try (ResultSet rs = stmt.executeQuery()) {
+                    int numRows = 0;
+                    while (rs.next()) {
+                        numRows++;
+                        dbVals.resultValue = getKitRequestShipping(rs);
+                    }
+                    if (numRows > 1) {
+                        throw new RuntimeException("Found " + numRows + " kits whit shortID " + hruid);
+                    }
+                }
+            } catch (SQLException ex) {
+                dbVals.resultException = ex;
+            }
+            if (dbVals.resultException != null) {
+                throw new RuntimeException("Error getting kit request with shortID " + hruid, dbVals.resultException);
+            }
+            return dbVals;
+        });
+        return Optional.ofNullable((KitRequestShipping) results.resultValue);
+    }
 
-
+    @Override
+    public Optional<ScanError> updateKitLabel(KitRequestShipping kitRequestShipping) {
+        Optional<ScanError> result = Optional.empty();
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(UPDATE_KIT_LABEL)) {
+                stmt.setString(1, kitRequestShipping.getKitLabel());
+                stmt.setLong(2, kitRequestShipping.getDsmKitId());
+                int rowsAffected = stmt.executeUpdate();
+                if (rowsAffected != 1) {
+                    dbVals.resultValue = new ScanError(kitRequestShipping.getDdpLabel(), "dsm_kit_id "
+                            + kitRequestShipping.getDsmKitId() + " does not exist or already has a Kit Label");
```

I think if you want the transaction to roll back in case of `rowsAffected > 1` you want to throw an exception here.

---

it will either be 0 or 1 so no need to roll back

---

and knowing how often scan error happen I don't want it to be throw one